### PR TITLE
Change implicit return

### PR DESCRIPTION
**Pull request by rdrpenguin04**

### CODE REVIEW

File: tests/full-example.ds [hunk 1]
```diff
@@ -12,17 +12,15 @@ struct vec3 {
 
 impl vec3 {
   fn length(): f32 {
```
Comment (jorolf) [inline tests/full-example.ds]: just saying but both methods could be converted to the new syntax

Comment (rdrpenguin04) [reply tests/full-example.ds]: I'm demonstrating both types of return.

File: tests/full-example.ds [hunk 2]
```diff
@@ -12,17 +12,15 @@ struct vec3 {
...
   
   fn to_str(): str {
     format("Vec3 {X: %d, Y: %d, Z: %d}", self.x, self.y, self.z);
   }
 }
 
-infix op +(a: vec3, b: vec3): vec3 {
-  vec3(a.x + b.x, a.y + b.y, a.z + b.z);
-}
+infix op +(a: vec3, b: vec3): vec3 = vec3(a.x + b.x, a.y + b.y, a.z + b.z);
```
Comment (jorolf) [inline tests/full-example.ds]: we have decided on ``->`` instead of `=`

File: tests/full-example.ds [hunk 3]
```diff
@@ -12,17 +12,15 @@ struct vec3 {
 
 impl vec3 {
   fn length(): f32 {
-    f32(sqrt(self.x*self.x + self.y*self.y + self.z*self.z));
+    return f32(sqrt(self.x*self.x + self.y*self.y + self.z*self.z));
   }
   
   fn to_str(): str {
     format("Vec3 {X: %d, Y: %d, Z: %d}", self.x, self.y, self.z);
```
Comment (jorolf) [inline tests/full-example.ds]: implicit return statements don't have a semicolon at the end iirc

Comment (rdrpenguin04) [reply tests/full-example.ds]: Fixed

Comment (jorolf) [reply tests/full-example.ds]: you fixed the wrong one 🤦‍♂️

Comment (rdrpenguin04) [reply tests/full-example.ds]: What do you mean? I removed the semicolon after the implicit return. There was only one. The one you're by is the explicit return, and that should have a semicolon iirc.

Comment (rdrpenguin04) [reply tests/full-example.ds]: Ah. Fixing...

File: tests/full-example.ds [hunk 4]
```diff
@@ -1,28 +1,24 @@
...
-  fn to_str(): str {
-    format("Vec3 {X: %d, Y: %d, Z: %d}", self.x, self.y, self.z);
-  }
+  fn to_str(): str -> format("Vec3 {X: %d, Y: %d, Z: %d}", self.x, self.y, self.z)
 }
 
-infix op +(a: vec3, b: vec3): vec3 {
-  vec3(a.x + b.x, a.y + b.y, a.z + b.z);
-}
+infix op +(a: vec3, b: vec3): vec3 -> vec3(a.x + b.x, a.y + b.y, a.z + b.z)
```
Comment (jorolf) [inline tests/full-example.ds]: the semicolon is still required here

Comment (rdrpenguin04) [reply tests/full-example.ds]: It's the same implicit return...

Comment (rdrpenguin04) [reply tests/full-example.ds]: Is it needed on both?

File: tests/full-example.ds [hunk 5]
```diff
@@ -1,32 +1,30 @@
...
+  fn format(x: str, ...): str;
+  fn printf(x: str, ...);
   fn sqrt(x: f64): f64;
 }
 
-struct vec3 {
-  x: f32,
-  y: f32,
-  z: f32
+struct vector3 {
```
Comment (jorolf) [inline tests/full-example.ds]: ``Vector3`` instead of ``vector3``

File: tests/full-example.ds [hunk 6]
```diff
@@ -1,36 +1,34 @@
...
+  fn printf(x: str, ...);
   fn sqrt(x: f64): f64;
 }
 
-struct vec3 {
-  x: f32,
-  y: f32,
-  z: f32
+struct Vector3 {
+  var x: f32,
```
Comment (jorolf) [inline tests/full-example.ds]: having mutable/immutable members in struct doesn't make much sense, does it?

**Pull request merged**